guillotina/tests/test_postgres.py

```diff
@@ -329,34 +331,54 @@ async def test_should_not_resolve_conflict_error_with_resolve(db, dummy_guilloti
         await cleanup(aps)
 
 
-@pytest.mark.skipif(DATABASE in ("cockroachdb", "DUMMY"), reason="Cockroach not support simple...")
-async def test_should_not_resolve_conflict_error_with_simple_strat(db, dummy_guillotina):
-    aps = await get_aps(db, "simple")
-    with TransactionManager(aps) as tm, await tm.begin() as txn:
-        ob1 = create_content()
-        ob2 = create_content()
-        txn.register(ob1)
-        txn.register(ob2)
+@pytest.mark.skipif(DATABASE in ("DUMMY",), reason="DUMMY not support simple...")
+async def test_should_raise_conflict_error_on_concurrent_insert(db, container_requester):
+    async with container_requester as requester:
+        with requester.db.get_transaction_manager() as tm:
+            txn = await tm.begin()
+            root_ob = await tm.get_root()
+            container = await create_container(root_ob, "test-container")
+            container.register()
+            await tm.commit(txn=txn)
 
-        await tm.commit(txn=txn)
+            txn1 = await tm.begin()
+            txn2 = await tm.begin()
 
-        # 1 started before 2
-        txn1 = await tm.begin()
-        txn2 = await tm.begin()
+            ob = create_content(parent=container)
+            ob_bis = create_content(id=ob.id, parent=container)
+            txn1.register(ob)
+            txn2.register(ob_bis)
 
-        ob1 = await txn1.get(ob1.__uuid__)
-        ob2 = await txn2.get(ob2.__uuid__)
+            await tm.commit(txn=txn2)
+            with pytest.raises(ConflictIdOnContainer):
+                await tm.commit(txn=txn1)
```

Comment:
The conflict is catch by the pg constraint but I think it's ok

Reply:
If we want to get the ConflictError because of TID mismatch, we would need to change the UPSERT query to something like:

```diff
INSERT INTO {{table_name}}
(zoid, tid, state_size, part, resource, of, otid, parent_id, id, type, json, state)
VALUES ($1::varchar({MAX_UID_LENGTH}), $2::bigint, $3::int, $4::int, $5::boolean,
        $6::varchar({MAX_UID_LENGTH}), $7::bigint, $8::varchar({MAX_UID_LENGTH}),
        $9::text, $10::text, $11::json, $12::bytea)
- ON CONFLICT (zoid)
+ ON CONFLICT ON CONSTRAINT {constraint_name}_parent_id_id_key
DO UPDATE SET
    tid = EXCLUDED.tid,
    state_size = EXCLUDED.state_size,
    part = EXCLUDED.part,
    resource = EXCLUDED.resource,
```